src/route/dataset-route.ts

```diff
@@ -0,0 +1,176 @@
+});
+
+apiRoute.get('/:dataset', async (req, res) => {
+    const datasetID = req.params.dataset;
+    if (datasetID === undefined || datasetID === null) {
+        res.status(404);
+        res.json({ message: 'Dataset not found... file is null or undefined' });
+        return;
+    }
+    const dataset = await Dataset.findOneBy({ id: datasetID });
+    if (!dataset) {
+        res.status(404);
+        res.json({ message: 'Dataset not found... dataset id is invalid' });
+        return;
+    }
+    const datafiles = await dataset.datafiles;
+    if (datafiles.length < 1) {
+        res.status(404);
+        res.json({ message: 'Dataset has no datafiles attached' });
+        return;
+    }
+    const dto = await datasetToDatasetDTO(dataset);
+    res.json(dto);
+});
+
+apiRoute.get('/:dataset/csv', async (req, res) => {
+    const dataLakeService = new DataLakeService();
+    const datasetID = req.params.dataset;
+    if (datasetID === undefined || datasetID === null) {
+        res.status(404);
+        res.json({ message: 'Dataset not found... You must specify a dataset ID' });
+        return;
+    }
+    const dataset = await Dataset.findOneBy({ id: datasetID });
+    if (dataset === undefined || dataset === null) {
+        res.status(404);
+        res.json({ message: 'Dataset not found... Dataset ID not found in Database' });
+        return;
+    }
+    const datafiles = await dataset.datafiles;
+    const fileToDownload: Datafile | undefined = datafiles
+        .sort(
+            (first: Datafile, second: Datafile) =>
+                new Date(second.creationDate).getTime() - new Date(first.creationDate).getTime()
+        )
+        .shift();
+    if (fileToDownload === undefined || fileToDownload === null) {
+        res.status(404);
+        res.json({ message: 'Dataset has no file attached' });
+        return;
+    }
+    const file = await dataLakeService.downloadFile(`${fileToDownload.id}.csv`);
+    if (file === undefined || file === null) {
+        res.status(404);
+        res.json({ message: 'File not found... file is null or undefined' });
+        return;
+    }
+    res.setHeader('Content-Length', file.length);
+    res.setHeader('Content-Type', 'text/csv');
+    res.setHeader('Content-Disposition', `attachment; filename=${fileToDownload.id}.csv`);
+    res.write(file, 'binary');
+    res.end();
+});
+
+apiRoute.get('/:dataset/xlsx', async (req, res) => {
+    const datasetID = req.params.dataset;
+    if (datasetID === undefined || datasetID === null) {
+        res.status(404);
+        res.json({ message: 'Dataset not found... You must specify a dataset ID' });
+        return;
+    }
+    const dataset = await Dataset.findOneBy({ id: datasetID });
+    if (dataset === undefined || dataset === null) {
+        res.status(404);
+        res.json({ message: 'Dataset not found... Dataset ID not found in Database' });
+        return;
+    }
+    res.json({
+        message: 'Not implmented yet'
+    });
+});
```

Comment:
Why is this a separate codepath from CSVs?
Blob upload/downloads are all the same, but only the mime-type differs and that should be supplied by the user?

Reply:
No they aren't... This endpoint should convert the requested Data File/Set into a proper XLSX file which is Microsofts proprietary XML/Binary format...  It's in the user needs to be able to provide this... I've been doing some experiments around this as time allows.

So we will need this endpoint... How it works or how our transformation processes work means we might just be streaming an already created file from the datalake or we might do it on the fly when the user requests it.